internal/extension/extension.go

```diff
@@ -191,6 +194,8 @@ func (em *ExtensionManager) SendEndInvocationRequest(ctx context.Context, functi
 	if err != nil || resp.StatusCode != 200 {
 		logger.Error(fmt.Errorf("could not send end invocation payload to the extension: %v", err))
 	}
+	io.Copy(io.Discard, resp.Body)
+	resp.Body.Close()
```

Comment:
These will fail in the case where resp is nil

Reply:
That's just a log message. Plus, we still want to read the body when status != 200.

Reply:
Oh sorry, I see what you mean. I think not.

I _believe_ the only time resp is nil is when err != nil. So we're basically already ensuring that resp != nil before checking the status code.